addons/binding/org.openhab.binding.fritzaha/src/main/java/org/openhab/binding/fritzaha/internal/discovery/FritzBoxBridgeDiscoveryParticipant.java

```diff
@@ -0,0 +1,105 @@
+ */
+package org.openhab.binding.fritzaha.internal.discovery;
+
+import static org.openhab.binding.fritzaha.FritzAhaBindingConstants.*;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.config.discovery.UpnpDiscoveryParticipant;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.jupnp.model.meta.DeviceDetails;
+import org.jupnp.model.meta.ModelDetails;
+import org.jupnp.model.meta.RemoteDevice;
+import org.openhab.binding.fritzaha.FritzAhaBindingConstants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link FritzBoxBridgeDiscoveryParticipant} is responsible for discovering
+ * new and removed FRITZ!Box devices. It uses the central
+ * {@link UpnpDiscoveryService}.
+ *
+ * @author Robert Bausdorf - Initial contribution
+ */
+public class FritzBoxBridgeDiscoveryParticipant implements
+		UpnpDiscoveryParticipant {
+	/**
+	 * Logger
+	 */
+	private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+	/**
+	 * Provide supported thing type uid's
+	 */
+	@Override
+	public Set<ThingTypeUID> getSupportedThingTypeUIDs() {
+		return FritzAhaBindingConstants.SUPPORTED_BRIDGE_THING_TYPES_UIDS;
+	}
+
+	/**
+	 * Create a discovery result from UPNP discovery
+	 */
+	@Override
+	public DiscoveryResult createResult(RemoteDevice device) {
+		ThingUID uid = getThingUID(device);
+		if (uid != null) {
+			logger.debug("discovered: " + device.getDisplayString());
+			Map<String, Object> properties = new HashMap<>(2);
+			properties.put(IP_ADDRESS, device.getDetails().getPresentationURI()
+					.getHost());
+			properties.put(SERIAL_NUMBER, device.getDetails()
+					.getPresentationURI());
+			DiscoveryResult result = DiscoveryResultBuilder
+					.create(uid)
+					.withProperties(properties)
+					.withLabel(device.getDetails().getFriendlyName())
+					.withRepresentationProperty(IP_ADDRESS)
+					.withTTL(
+							Math.max(MIN_MAX_AGE_SECS, device.getIdentity()
+									.getMaxAgeSeconds())).build();
+			return result;
+		} else {
+			logger.debug("no uid for: " + device.getDisplayString());
+			return null;
+		}
+	}
+
+	/**
+	 * Compute a FRITZ!Box thind UID.
+	 */
+	@Override
+	public ThingUID getThingUID(RemoteDevice device) {
+		DeviceDetails details = device.getDetails();
+		if (details != null) {
+			logger.debug(details.getFriendlyName() + ": "
+					+ details.getPresentationURI());
+			ModelDetails modelDetails = details.getModelDetails();
+			if (modelDetails != null) {
+				String modelName = modelDetails.getModelName();
+				logger.debug("ModelName: " + modelName);
+				if (modelName != null) {
+					if (modelName.startsWith(BRIDGE_MODEL_NAME)) {
+						return new ThingUID(AHABRIDGE_THING_TYPE, details
+								.getPresentationURI().getHost()
```

Comment:
I actually had the same, I had different label texts for both results: The additional one was a "MediaRenderer" or something like that, so the DLNA features of the FRITZ!Box. This is a similar problem as I have seen with the Samsung TV (https://github.com/openhab/openhab2/pull/106#issuecomment-87270807). Difficult to say what is the best way to deal with it. Maybe one should treat this indeed as two separate devices as possibly completely different bindings will be involved (e.g. an DLNA-binding for the other UDN). What do you think?